tests/lax_numpy_operators_test.py

```diff
@@ -731,6 +731,26 @@ def testSpacingSubnormals(self, dtype):
     self._CheckAgainstNumpy(np.spacing, jnp.spacing, args_maker, check_dtypes=True, tol=0)
     self._CompileAndCheck(jnp.spacing, args_maker, tol=0)
 
+  @jtu.ignore_warning(category=RuntimeWarning, message="divide by zero.*")
+  @jtu.ignore_warning(category=RuntimeWarning, message="invalid value.*")
+  def testFloorDivideByZeroInt(self):
+    x1 = jnp.eye(3, dtype=int)
+    x2 = jnp.zeros((3, 3), dtype=int)
+    jnp_result = jnp.floor_divide(x1, x2)
+    np_result = np.floor_divide(x1, x2)
+    self.assertAllClose(jnp_result, np_result, check_dtypes=True)
+
+  @jtu.ignore_warning(category=RuntimeWarning, message="divide by zero.*")
+  @jtu.ignore_warning(category=RuntimeWarning, message="invalid value.*")
+  def testFloorDivideFloatSpecialCases(self):
+    x = jnp.array([-np.nan, -np.inf, -2., -1., -0., 0., 1., 2., np.inf, np.nan])
+    x1 = x[:, None]
+    x2 = x[None, :]
+    jnp_result = jnp.floor_divide(x1, x2)
+    np_result = np.floor_divide(x1, x2)
+    self.assertAllClose(jnp_result, np_result, check_dtypes=True)
+    self.assertAllClose(jnp.signbit(jnp_result), jnp.signbit(np_result), check_dtypes=True)
```

Comment:
It appears that this assertion is failing on some platforms. I'm not sure why:
```
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 2 / 100 (2%)
 ACTUAL: array([[ True,  True,  True,  True,  True,  True,  True,  True,  True,
         True],
       [False, False, False, False, False,  True, False, False, False,...
 DESIRED: array([[ True,  True,  True,  True,  True,  True,  True,  True,  True,
         True],
       [False, False, False, False, False,  True, False, False, False,...
```

Reply:
I think the problem, here is the usage of `jnp.signbit` to test the sign of the result. In NumPy, the signbit of `nan` resulted in some cases is set even though the signbit of both the inputs is not set.

```python
>>> np.signbit([np.inf, 1.])
array([False, False])
>>> np.signbit(np.floor_divide(np.inf, 1.))
<ipython-input-52-d8652a8e4b0b>:1: RuntimeWarning: invalid value encountered in floor_divide
  np.signbit(np.floor_divide(np.inf, 1.))
True
>>> jnp.signbit(floor_divide(np.inf, 1.))
Array(False, dtype=bool)
```
We can avoid this by using `lax.sign` instead of `jnp.signbit`.

Reply:
Changing to `lax.sign` will not correctly differentiate between `-0.0` and `0.0`, or between `-NaN` and `NaN`, which as I understand it is the only point of this check. If we change to `lax.sign`, then this test becomes unnecessary, because the sign match is already covered by the equality check one line above.

So I'd say, either we figure out how to make the `signbit` version work, or we remove the check altogether. What do you think?

Reply:
I could not find a way to resolve this. I have tested it on colab CPU and Mac M1 Pro chip. The tests does not fail in these environments with NumPy 1.26.4, 2.1.3 and 2.2.0rc1 versions.